server/sonar-web/src/main/js/apps/marketplace/Search.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { debounce } from 'lodash';
+import Tooltip from '../../components/controls/Tooltip';
+import { Query } from './utils';
+import { translate } from '../../helpers/l10n';
+
+interface Props {
+  query: Query;
+  updateCenterActive: boolean;
+  updateQuery: (newQuery: { [key: string]: any }) => void;
```

Comment:
should be `newQuery: Query`

Reply:
well, yes and no... because I want to be able to pass down an incomplete Query, since then the `updateQuery` merge the existing query with the new incomplete one. Kind of like `setState`, how is setState typed ? I should try to type it the same!

Reply:
Ok I'll use `Partial<Query>`